lib/Runtime/Library/JavascriptFunction.cpp

```diff
@@ -691,10 +697,10 @@ namespace Js
         __finally
         {
             // 0xE06D7363 is C++ exception code
-            if (exceptionCode != 0 && !IsDebuggerPresent() && exceptionCode != 0xE06D7363 && exceptionAction != EXCEPTION_CONTINUE_EXECUTION)
+            if (AbnormalTermination() && exceptionCode != 0 && !IsDebuggerPresent() && exceptionCode != 0xE06D7363)
```

Comment:
I looked at implementation and seems not that expensive. I'll swap the condition around though so we only come here if exceptionCode != 0